<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>Home</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Home</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#finalizing">Finalizing</a></li>
<li><a href="#process-sequence-data">Process sequence data</a></li>
<li><a href="#analysis">Analysis</a></li>
<li><a href="#download-data">Download data</a></li>
<li><a href="#one-time-investigations">One-time investigations</a></li>
<li><a href="#other">Other</a></li>
<li><a href="#lcl-data-from-a-full-flowcell">LCL data from a full flowcell</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-11-08</p>
<p><strong>Code version:</strong> bd286a36f14d3b332285cdc7e62258b1f616bb14</p>
<p>We used this site to collaborate and share our results. Please feel free to explore. The results that made it into the final paper are in the section <strong>Finalizing</strong> below. Here are some useful links:</p>
<ul>
<li><a href="../paper/tung-et-al.docx">Paper</a></li>
<li>GEO record <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77288">GSE77288</a> for the raw FASTQ files</li>
<li><a href="https://github.com/jdblischak/singleCellSeq">GitHub repo</a> for code and data</li>
</ul>
<div id="finalizing" class="section level2">
<h2>Finalizing</h2>
<ul>
<li><a href="process-samples.html">Process sequencing data - From fastq to gene counts</a></li>
<li><a href="prepare-counts.html">Create gene-x-sample count matrices and annotation files</a></li>
<li><a href="qc-filter-ipsc.html">Quality control of single cells</a></li>
<li><a href="filter.html">Filter genes and low-quality single cells</a></li>
<li><a href="qc-plots.html">Quality control of UMIs and ERCC spike-ins</a></li>
<li><a href="compare-reads-v-molecules-per-batch-test.html">Compare reads to molecules counts</a></li>
<li><a href="subsample-final.html">Subsampling</a> (<a href="subsample-lcl-final.html">LCL</a>, <a href="subsample-final-rep.html">per batch</a>)</li>
<li><a href="cpm.html">Standardize with counts per million</a></li>
<li><a href="poisson-glm-normalization.html">Account for technical noise measured with ERCC controls using Poisson GLM transformation</a></li>
<li><a href="batch-correction.html">Mixed effect model for batch correction</a></li>
<li><a href="data-transformation.html">Visualization of data transformations</a></li>
<li><a href="cv-adjusted-summary-pois-final.html">Comparing cell-to-cell variation between individuals</a> (<a href="paper-additional-analysis.html">Additional figures</a>)</li>
</ul>
</div>
<div id="process-sequence-data" class="section level2">
<h2>Process sequence data</h2>
<ul>
<li><a href="process-samples.html">Pipeline</a> - (<a href="process-lcl.html">LCLs</a>)</li>
<li><a href="total-counts.html">Total counts</a></li>
<li><a href="qc-by-lane.html">Quality control at level of sequencing lane</a></li>
<li><a href="sum-counts-per-sample.html">Sum counts per sample</a></li>
<li><a href="qc-cell-ipsc.html">Quality control at level of single cell samples</a> - (<a href="qc-cell-lcl.html">LCLs</a>)
<ul>
<li><a href="depth.html">Sequencing depth and cellular RNA content</a> - (<a href="depth-final.html">Final</a>)</li>
<li><a href="qc-filter-ipsc.html">Attempt to finalize quality control</a></li>
</ul></li>
</ul>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<ul>
<li><a href="noisy-gene.html">Identification of noisy genes</a></li>
<li><a href="compare-reads-v-molecules.html">Compare read versus molecule counts</a> - (<a href="compare-reads-v-molecules-per-batch.html">per batch</a>,<a href="compare-reads-v-molecules-per-batch-test.html">per batch conversion rates comparison</a>, <a href="compare-reads-v-molecules-in-cells.html">per cell</a>)
<ul>
<li><a href="islam2014.html">Islam et al., 2014</a></li>
<li><a href="vallejos2015.html">Vallejos et al., 2015</a></li>
</ul></li>
<li><a href="variance.html">Variance within and between individaul</a></li>
<li><a href="ercc-correlation.html">Correlation with ERCC spike-ins</a>
<ul>
<li><a href="ercc-counts.html">Counts of each ERCC in single cell</a></li>
<li><a href="capture-efficiency.html">Capture efficiency</a></li>
<li><a href="ercc-over-sequenced.html">Over-sequenced ERCC in LCLs</a></li>
<li><a href="ercc-cpm.html">cmp ERCC or not</a></li>
</ul></li>
<li><a href="ercc-normalization.html">ERCC normalization</a>
<ul>
<li><a href="gamma-regression-model.html">Gamma regression model (GRM)</a></li>
<li><a href="basics.html">BASiCS</a>
<ul>
<li><a href="basics-wo-batch.html">without modeling batch</a></li>
<li>Using too few iterations gives a poor fit. These are older versions of the analysis kept to demonstrate this(<a href="basics-4000.html">4000 iterations</a>, <a href="basics-20000.html">20000 iterations</a>)</li>
</ul></li>
<li><a href="shrunk-linear-transform.html">Shrunk linear transformation</a></li>
<li><a href="basics-variance-components.html">Variance components</a></li>
<li><a href="poisson-glm-normalization.html">Poisson GLM normalization</a></li>
</ul></li>
<li><a href="subsample.html">Subsample</a> - (<a href="subsample-lcl.html">LCLs</a>)
<ul>
<li><a href="correlate-single-to-bulk.html">Correlation of single and bulk cells</a></li>
<li><a href="detect-genes.html">Number of expressed genes</a> - (<a href="detect-genes-lcl.html">LCLs</a>)</li>
<li><a href="subsample-high-coverage-lcl.html">High coverage single cells</a></li>
<li><a href="subsample-variance.html">Cell to cell variance</a></li>
</ul></li>
<li><a href="cell-cycle.html">Cell-cycle analysis</a> - (<a href="cell-cycle-post-correction.html">final</a>), (<a href="cell-cycle-investigate.html">cycle-gene-set</a>)<br /></li>
<li><a href="stemness.html">pluripotency gene expression</a></li>
<li><a href="pca-correlation-rafa.html">Proportion of detected genes</a> - (<a href="pca-correlation-rafa-filter.html">filter and final</a>)
<ul>
<li><a href="gene-and-total-molecule.html">Detected genes and total molecule counts</a></li>
<li><a href="prop-genes-detected.html">More exploring of the proportion of genes detected</a> - <a href="prop-genes-detected-unfiltered.html">Using all cells</a></li>
</ul></li>
<li><a href="batch-limma.html">Batch effect correction with mixed modeling</a> - (<a href="batch-limma-final.html">linear transformation, filter and final</a>, <a href="batch-limma-poisson.html">Poisson transformation, filter and final</a>)
<ul>
<li><a href="batch-limma-final-per-gene.html">Per gene correction (filtered and final)</a></li>
</ul></li>
<li><a href="pca-individual.html">Individual PCAs</a></li>
<li>Ordering effect of capture sites
<ul>
<li>All genes (<a href="oefinder-analysis.html">DESeq-normalized</a>, <a href="oefinder-analysis-cpm.html">CPM-normalized</a>)</li>
<li><a href="oefinder-analysis-ercc.html">ERCC genes</a></li>
<li><a href="oefinder-analysis-cpm-batch.html">Endogeneous genes per batch, per individual</a></li>
</ul></li>
<li><a href="gene-symbol.html">Extract gene symbols</a></li>
<li>Cell-to-cell variation analysis
<ul>
<li><a href="cv-sparsity.html">CV and sparsity</a></li>
<li>Mean-adjusted CV – ( <a href="cv-adjusted-wo-19098-r2.html">Normalize CV excluding outlier batch</a>, <a href="cv-adjusted.html">Normalize CV including outlier batch</a>, <a href="cv-adjusted-profile.html">Gene rankings</a>, <a href="cv-adjusted-profile-annotation.html">Annotations</a>, <a href="cv-adjusted-comparison.html">Gene-level dissimilarity measure</a> )</li>
<li>Compare mean-adjusted CVs – (<a href="cv-adjusted-all-genes-wo-19098-2.html">ANOVA</a>, <a href="cv-annotation.html">ANOVA &amp; annotations</a>, <a href="cv-adjusted-statistical-test.html">Sum of Squared Deviation from the Median</a>, <a href="cv-adjusted-statistical-test-permute.html">Permutation-based p-values</a> )</li>
<li>Noisy genes analysis – (<a href="cv-adjusted-summary.html">final and Gaussian-based transformation</a>, <a href="cv-adjusted-summary-pois.html">final and Poisson-based transformation</a>, <a href="compare-distribution-pluripotency.html">Pluripotent gene density plots</a> )</li>
<li>Putting it all together… (<a href="cv-adjusted-summary-pois.html">All cells</a>, <a href="cv-adjusted-summary-pois-expressed.html">Expressed cells</a>, <a href="cv-adjusted-summary-pois-dropout.html">Undetected cells</a> )</li>
</ul></li>
<li><a href="qc-plots.html">Quality control plots</a></li>
<li><a href="total-molecule.html">Total molecule-counts and standardization</a></li>
<li>Exploring sequence coverage
<ul>
<li>Endogenous genes
<ul>
<li><a href="coverage-endogenous.html">molecules</a></li>
<li><a href="coverage-endogenous-reads.html">reads</a></li>
<li><a href="coverage-endogenous-bulk.html">bulk</a></li>
</ul></li>
<li>ERCC spike-ins
<ul>
<li><a href="coverage-ercc.html">molecules</a></li>
<li><a href="coverage-ercc-reads.html">reads</a></li>
<li><a href="coverage-ercc-bulk.html">bulk</a></li>
</ul></li>
<li>Using ngsplot
<ul>
<li><a href="ngsplot-setup.html">Setup</a></li>
<li><a href="ngsplot-endogenous.html">endogenous genes</a></li>
<li><a href="ngsplot-ercc.html">ERCC genes</a></li>
</ul></li>
</ul></li>
<li><a href="pca-quantification.html">PCA quantification</a></li>
</ul>
</div>
<div id="download-data" class="section level2">
<h2>Download data</h2>
<ul>
<li><a href="download_flow_cell_01.html">Download flow cell 01 - C6WYKACXX</a></li>
<li><a href="download_flow_cell_02.html">Download flow cell 02 - C6WURACXX</a></li>
<li><a href="download_flow_cell_03.html">Download flow cell 03 - C723YACXX</a></li>
<li><a href="download_flow_cell_04.html">Download flow cell 04 - C72JMACXX</a></li>
</ul>
</div>
<div id="one-time-investigations" class="section level2">
<h2>One-time investigations</h2>
<ul>
<li><a href="verify-md5sum.html">Verify md5 checksums</a></li>
<li><a href="files-for-testing.html">Small and large files for testing</a></li>
<li><a href="bug-conversion-01.html">Tracking down bug in reads to molecules conversion - Attempt 01</a></li>
<li><a href="read2pos.html">The effect of using featureCounts read2pos flag</a></li>
<li><a href="bug-conversion-02.html">Tracking down bug in reads to molecules conversion - Attempt 02</a></li>
<li><a href="bug-conversion-03.html">Tracking down bug in reads to molecules conversion - Attempt 03</a></li>
<li><a href="bug-conversion-04.html">Tracking down bug in reads to molecules conversion - Attempt 04</a></li>
</ul>
</div>
<div id="other" class="section level2">
<h2>Other</h2>
<ul>
<li><a href="analysis.html">Assessment of rapid run flow cell</a></li>
</ul>
</div>
<div id="lcl-data-from-a-full-flowcell" class="section level2">
<h2>LCL data from a full flowcell</h2>
<ul>
<li><a href="qc-cell-library.html">Quality control of cell/library</a></li>
</ul>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
