---
title: "Total molecule counts"
author: "Po-Yuan Tung"
date: 2016-04-18
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 12)
```

Recreate the [Sequencing depth and cellular RNA content](depth.html) using updated data files to make figures for the paper.

## Setup

```{r packages, message=FALSE}
library("dplyr")
library("tidyr")
library("edgeR")
library("lmtest")
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 12))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())
source("functions.R")
```

Input annotation 

```{r input-annotation}
anno_single <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_single)
```

Input single cell observational quality control data.

```{r input-qc}
qc <- read.table("../data/qc-ipsc.txt", header = TRUE,
                 stringsAsFactors = FALSE)
qc <- qc %>% arrange(individual, replicate, well)
stopifnot(qc$individual == anno_single$individual,
          qc$replicate == anno_single$replicate,
          qc$well == anno_single$well)
head(qc)
```

Incorporate informatin on cell number, concentration, and TRA1-60 status.

```{r add-qc}
anno_single$cell_number <- qc$cell_number
anno_single$concentration <- qc$concentration
anno_single$tra1.60 <- qc$tra1.60
```

Keep only quality single cell

```{r quality-cell}
quality_single_cells <- scan("../data/quality-single-cells.txt",
                             what = "character")
anno_single <- anno_single[anno_single$sample_id %in% quality_single_cells,]
```

Input molecule counts after filtering

```{r input-molecule-counts}
molecules <- read.table("../data/molecules-filter.txt", header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(colnames(molecules) == anno_single$sample_id)
ercc_index <- grepl("ERCC", rownames(molecules))
anno_single$total_molecules_gene = colSums(molecules[!ercc_index, ])
anno_single$total_molecules_ercc = colSums(molecules[ercc_index, ])
anno_single$total_molecules = colSums(molecules)
anno_single$num_genes = apply(molecules[!ercc_index, ], 2, function(x) sum(x > 0))
```


## The total molecule-counts

For wells with a single cell observed, the total molecule-counts range from
`r format(min(anno_single$total_molecules_gene[anno_single$cell_number == 1]))`
to
`r format(max(anno_single$total_molecules_gene[anno_single$cell_number == 1]))`,
the first quartile for the total number of gene molecules is
`r format(quantile(anno_single$total_molecules_gene[anno_single$cell_number == 1], probs = 0.25), scientific = FALSE)`,
and the third quartile is
`r format(quantile(anno_single$total_molecules_gene[anno_single$cell_number == 1], probs = 0.75), scientific = FALSE)`.

```{r total_molecules, fig.width=12, fig.height=6}
## create a color palette with one color per individual and different shades for repplicates
great_color_8 <- c("#CC3300", "#FF9966", "#006633", "#009900", "#99FF99", "#3366FF", "#6699FF", "#66CCFF")

plot_total_molecules_gene <- ggplot(anno_single,
                       aes(x = as.factor(batch), y = total_molecules_gene / 10^3, fill = as.factor(batch))) + 
                       geom_boxplot(alpha = .01, width = .2) +
                       geom_violin(alpha = .5) + 
                       scale_fill_manual(values = great_color_8) +
                       labs(x = "Batch", y = "Total gene molecule-counts (10^3)") +
                       theme(axis.text.x = element_text(hjust=1, angle = 45))

plot_total_molecules_ercc <-  plot_total_molecules_gene %+%
                                   aes(y = total_molecules_ercc / 10^3) +
                                   labs(y = "Total ERCC molecule-counts (10^3)")

summary(aov(total_molecules_gene ~ individual, data = anno_single))

plot_grid(plot_total_molecules_gene + theme(legend.position = "none"),
          plot_total_molecules_ercc + theme(legend.position = "none"),
          labels = LETTERS[1:2])
```

## The relationship between concentration and total molecule-counts (not including NA19098)

As we try to understand the general relationships between sequencing results and cellular mRNA content, we remove outlier batches.
NA19098 replicate 1 failed the quantification of the concentration of the single cells and was hence removed. Because NA19098 concentration is only quantified in one replicate, we removed NA19098 from analysis involving batch differences and concentration.

```{r filter-outliers}
anno_single_6 <- anno_single %>% filter(individual != "NA19098")
```

```{r molecules-num-cells-batch, fig.width=12, fig.height=8}
## look at endogenous genes
plot_conc_molecules_gene_individual <- 
  ggplot(anno_single_6,
         aes(x = concentration, 
             y = total_molecules_gene / 10^3, color = individual)) +
     geom_point(alpha = 0.8) +
     scale_colour_manual(values = c("#33CC66", "#6699FF")) +
     geom_smooth(method = "lm", se = FALSE) +
     labs(x = "Concentration (ng/ul)", y = "Total gene molecule-counts (10^3)")

## look at ERCC spike-ins
plot_conc_molecules_ercc_individual <-
  plot_conc_molecules_gene_individual %+%
     aes(y = total_molecules_ercc / 10^3) +
     labs(y = "Total ERCC molecule-counts (10^3)")

## plots
plot_grid(plot_conc_molecules_gene_individual + theme(legend.position = "none"), 
          plot_conc_molecules_ercc_individual + theme(legend.position = "none"),
          labels = LETTERS[1:2])

## Is there a difference across the three individuals
table(anno_single_6$individual, anno_single_6$replicate)
fit0 <- lm(total_molecules_gene ~ concentration, 
   data = anno_single_6)
fit1 <- lm(total_molecules_gene ~ concentration + as.factor(individual), 
   data = anno_single_6)
# use likelihood ratio test to detect individual differences
lrtest(fit1, fit0)
```


Calculate correlation of concentration and molecule counts

```{r cor}
## for each individual
for (i in 1:length(unique(anno_single_6$individual))) {
    print(unique(anno_single_6$individual)[i])
    select_individual <- with(anno_single_6, individual == unique(individual)[i])
    print(cor(anno_single_6[select_individual,7],anno_single_6[select_individual,9]))
}

## for each batch
for (i in 1:length(unique(anno_single_6$batch))) {
    print(unique(anno_single_6$batch)[i])
    select_replicate <- with(anno_single_6, batch == unique(batch)[i])
    print(cor(anno_single_6[select_replicate,7],
              anno_single_6[select_replicate,9]))
}
```


## ERCC counts and total molecule counts

```{r ercc-total-counts}
## calulate ERCC percentage
anno_single <- anno_single %>%
  mutate(perc_ercc_molecules = total_molecules_ercc / total_molecules * 100)

## ERCC molecule versus total molecule
plot_gene_mol_ercc_mol <- ggplot(anno_single,
                                 aes(x = total_molecules_gene,
                                     y = total_molecules_ercc)) +
                         geom_point(aes(color = individual, shape = replicate, alpha = 0.8))+
                         labs(x = "Total gene molecules", y = "Total ERCC molecules") 

plot_gene_mol_perc_ercc <- plot_gene_mol_ercc_mol %+%
                           aes(x = perc_ercc_molecules)+
                           labs(x = "Percent ERCC molecules")

plot_grid(plot_gene_mol_ercc_mol + theme(legend.position = "none"),
          plot_gene_mol_perc_ercc + theme(legend.position= "none"),
          labels = LETTERS[1:2])

## Is there a difference across the three individuals
table(anno_single$individual, anno_single$replicate)
fit0 <- lm(total_molecules_ercc ~ 1,  data = anno_single)
fit1 <- lm(total_molecules_ercc ~ 1 + as.factor(individual), data = anno_single)
anova(fit0, fit1)

## Is there a difference across replicates for all individuals
table(anno_single$batch, anno_single$individual)
fit2 <- lm(total_molecules_gene ~ concentration + as.factor(individual) + batch, 
   data = anno_single)

summary(fit2)$coef

lrtest(fit1,fit2)

```

## Plots for paper 
```{r plot,fig.width=12, fig.height=8}
plot_grid(plot_total_molecules_gene + theme(legend.position = "none"),
          plot_total_molecules_ercc + theme(legend.position = "none"),
          plot_conc_molecules_gene_individual + theme(legend.position= "none"),
          plot_gene_mol_ercc_mol + theme(legend.position = "none"),
          labels = LETTERS[1:4])
```

## Expression profile
```{r expression-profile,fig.width=12, fig.height=6}
## pca of endogenous genes
pca_molecules_profile_ENSG <- run_pca(molecules[!ercc_index, ])
pca_molecules_ENSG_plot <- plot_pca(pca_molecules_profile_ENSG$PCs,
         explained = pca_molecules_profile_ENSG$explained,
         metadata = anno_single, color = "individual",
         shape = "replicate", alpha = 0.5, size = 2.2) +
  labs(title = "Endogenous gene expression profile")


## pca of ERCC spike-ins
pca_molecules_profile_ERCC <- run_pca(molecules[ercc_index, ])
pca_molecules_ERCC_plot <- plot_pca(pca_molecules_profile_ERCC$PCs,
         explained = pca_molecules_profile_ERCC$explained,
         metadata = anno_single, color = "individual",
         shape = "replicate", alpha = 0.5, size = 2.2) +
  labs(title = "ERCC spike-in expression profile")

## make plots
plot_grid(pca_molecules_ENSG_plot + theme(legend.position = "none"),
          pca_molecules_ERCC_plot + theme(legend.position = "none"),
          labels = LETTERS[1:2])

```


## Session information

```{r info}
sessionInfo()
```
